go.mod

```diff
@@ -3,15 +3,17 @@ module github.com/elastic/package-registry
 go 1.18
 
 require (
+	cloud.google.com/go/storage v1.22.0
 	github.com/Masterminds/semver/v3 v3.1.0
 	github.com/elastic/go-licenser v0.3.1
 	github.com/elastic/go-ucfg v0.8.4-0.20200415140258-1232bd4774a6
 	github.com/felixge/httpsnoop v1.0.2
+	github.com/fsouza/fake-gcs-server v1.37.12
```

Comment:
I am wondering if we need a full gcs implementation. Files required by the registry are available in public buckets, right? Could we just use `httptest` serving static files for testing? Is there any file only available in private buckets?

Reply:
Well, the index itself is private, right? Could we read it with authenticated HTTP requests so we don't need a GCP client in the registry?

Reply:
> If we want to go with authenticated HTTP requests, then we need a GCP signer code in the Package Registry, right?

> Sample of URL signing in Go: https://gist.github.com/poindextrose/afa4d4bb5e93c4c3de77. It means that we will still need a dependency on `google.golang.org/cloud/storage`.

Umm, are signed urls the only option to download private files?

It seems to be possible to use normal urls and a bearer token (seen [here](https://cloud.google.com/storage/docs/authentication) and [here](https://stackoverflow.com/questions/64781844/google-cloud-storage-downloading-a-file-using-curl-request)). We would only need a dependency to get this token, but then everything else could be normal http.

Reply:
Correct me if I'm wrong, but if this is OAuth, then it may happen that the token expires while the Package Registry is running and we will have to reauthorize again?